### ISSUE TITLE
Rule suggestion: Warn if a using declaration is followed immediately by a loose block

### ISSUE THREAD

**Issue by PathogenDavid** [dotnet/roslyn-analyzers#4899]
This rule is inspired by a dumb mistake I made recently that I thought might make a good candidate for an analyzer.

### Describe the problem you are trying to solve

At a glance, the following (semi-contrived) example looks correct and sensible. The intent is that the `StreamWriter` is disposed on the line marked with `<--`.

```csharp
static void TestFunction()
{
    using StreamWriter f = new("HelloWorld.txt");
    {
        f.WriteLine("Hello, world!");
    } // <--

    Console.WriteLine(File.ReadAllText("HelloWorld.txt"));
}
```

However, the author accidentally used a [using declaration](https://github.com/dotnet/csharplang/blob/93404fdb7387962a80cdc3e3a5e51daacd89c2d9/proposals/csharp-8.0/using.md) where they clearly intended to use a using statement. As such the `StreamWriter` is disposed at the end of `TestFunction`, which causes the `ReadAllText` to fail at runtime due to the file still being locked.

They should've written this instead:

```csharp
static void TestFunction()
{
    using (StreamWriter f = new("HelloWorld.txt"))
    {
        f.WriteLine("Hello, world!");
    } // <--

    Console.WriteLine(File.ReadAllText("HelloWorld.txt"));
}
```

### Describe suggestions on how to achieve the rule

If one or more sequence of `LocalDeclarationStatement`s representing using declarations are followed immediately by a `Block`, the rule fails. (Potentially with a heuristic involving the trivia involved to avoid cases where the block is more likely for code organization and just happens to follow a using declaration.)

The suggested code fix would be to rewrite the using declarations + block into a `UsingStatement` (as shown above.)

### Additional context

Obviously the "wrong" example above is technically valid C#, but I couldn't think of a real scenario where someone would want code shaped that way without the intent being to use a using statement.

I also felt like this would be really easy to overlook in a code review in a scenario where the important of the `using` scope was less important and wouldn't necessarily blow up immediately at runtime.

In my case, it was also a non-trivial custom type rather than `StreamWriter` so I accidentally went down a rabbit hole of trying to debug my `Dispose` implementation. (I probably would've noticed the issue sooner if it was just `StreamWriter`.)

Definitely open to hear I'm just bad though 😅

**Comment by Youssef1313** [comment]
I remember doing a similar mistake that wasn't in a `using` (but mine was more obvious). Does it make sense to have a more generalized approach for such analyzer (but shouldn't be *too* general)

```csharp
using System.Collections.Generic;

public class C
{
    public void M()
    {
        var x = new List<string>();
        {
            GetString();
        };
    }
 
    private string GetString()
    {
        return "";
    }
}
```

**Comment by PathogenDavid** [comment]
@Youssef1313 That actually made me think of a situation where my original problem code might come up legitimately. It's a semi-common pattern in my own code to use a block for non-trivial initialization of an object. ([Random example of my own here](https://github.com/InfectedLibraries/Biohazrd/blob/333eb67e7302ec67ce70708d9e4cd15a47febd7e/Biohazrd/TranslatedLibraryConstantEvaluator.cs#L277-L278))

While I'd consider using that pattern in combination with a using declaration to be a bad idea due to the possible confusion with a using statement, I could see someone doing it intentionally and having this theoretical analyzer be a false positive.

**Comment by Youssef1313** [comment]
In these cases I think it's clearer to leave an empty line before the opening brace. This should satisfy the analyzer, keep the intentional braces as them, and also make the code more clear.

**Comment by mavasani** [comment]
We agree with @Youssef1313's [suggestion](https://github.com/dotnet/roslyn-analyzers/issues/4899#issuecomment-785832011) is the preferred approach. We should create a new rule in Roslyn repo for block should be preceded by a blank line.

**Comment by Youssef1313** [comment]
> We agree with @Youssef1313's [suggestion](https://github.com/dotnet/roslyn-analyzers/issues/4899#issuecomment-785832011) is the preferred approach. We should create a new rule in Roslyn repo for block should be preceded by a blank line.

Opened #51668 in Roslyn.
